File: Spring/src/main/java/Service/Imagen_Service.java
package Service;

import java.util.List;
import Model.Imagen;
import Model.Producto;

public interface Imagen_Service {
	
	public boolean 		save  				(Imagen i);
	public boolean 		update				(Imagen i);
	public boolean 		delete				(Imagen i);
	
	public List<Imagen> getAll						();
	public Imagen 		getById				(Imagen i);
	
	public List<Imagen> producto 		  (Producto p);
}

File: Angular/src/app/service/imagen.service.ts
import { Producto } from './../model/producto';
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Imagen } from '../model/imagen';

@Injectable({
  providedIn: 'root'
})
export class ImagenService {

  private baseUrl = `${window.location.protocol}//${window.location.hostname}:8080/api/imagen-`;

  constructor(private http:HttpClient) { }

  getByAllByProduct(producto:Producto):Observable<any>
  {
    return this.http.post(`${this.baseUrl}producto`, producto);
  }


  imagenUrl(imagen:Imagen):string
  {
    return `${this.baseUrl}mostrar/${imagen.id}`
  }

  imagenUrlUrl(imagen:Imagen):string
  {
    return `${this.baseUrl}mostrar-url/${imagen.id}`
  }

  getAll():Observable<any>
  {
    return this.http.get(`${this.baseUrl}list`);
  }
}
